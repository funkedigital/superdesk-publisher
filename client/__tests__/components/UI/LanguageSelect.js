import React from "react";
import LanguageSelect from "../../../components/UI/LanguageSelect";
import { render, fireEvent, getByText } from "@testing-library/react";

const languages = [
  { qcode: 'pl', name: 'Polish' },
  { qcode: 'en', name: 'English' },
  { qcode: 'de', name: 'German' },
]

describe("UI/LanguageSelect", () => {
  it("renders correctly", () => {
    const { container, getByText } = render(
      <LanguageSelect languages={languages} selectedLanguageCode="en" setLanguage={jest.fn()} />
    );

    expect(getByText("Polish")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("setLanguage function fired", () => {
    const setLanguage = jest.fn();
    const { getByText } = render(
      <LanguageSelect languages={languages} selectedLanguageCode="en" setLanguage={setLanguage} />
    );
    const button = getByText("Polish");

    fireEvent.click(button);
    expect(setLanguage).toHaveBeenCalled();
    expect(setLanguage).toHaveBeenCalledWith("pl");
  });
});
