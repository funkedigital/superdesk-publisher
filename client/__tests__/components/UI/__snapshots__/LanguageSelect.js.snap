// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/LanguageSelect renders correctly 1`] = `
<div
  class="subnav__spacer subnav__spacer--no-margin"
/>
`;
